api/poetry.lock

```diff
@@ -3293,6 +3293,21 @@ pytest = ">=5.0"
 [package.extras]
 dev = ["pre-commit", "pytest-asyncio", "tox"]
 
+[[package]]
+name = "pytest-testmon"
+version = "2.0.13"
+description = "selects tests affected by changed files and methods"
```

Comment:
I understand this only runs a subset of the tests that we have. Does this PR make that only possible for comparisons to main or does the whole CI run during the initial PR creation?

Reply:
This code uses the latest version of the base branch, main in this case, as a baseline for comparison, to run only tests related to changed code files. So every time you create a PR, or modify it, only tests affected by changes will run. Once a PR is merged into main, we will create a new baseline for comparisons.